src/ZArray.jl

```diff
@@ -361,20 +363,12 @@ function filterfromtype(::Type{<:AbstractArray{T}}) where T
   (VLenArrayFilter{T}(),)
 end
 
+filterfromtype(::Type{<:Union{<:AbstractString, Union{<:AbstractString, Missing}}}) = (VLenUTF8Filter(),)
```

Comment:
Here we would default to using VLenUTF8. Do you happen to know how this is handled in the python implementation? If it the default string representation there as well, I agree with this change, just some years back I remember hitting a lot of MaxLengthStrings.

Reply:
I think that's because numpy by default stores string arrays as fixed-length strings. The default string representation for Python string objects is [vlen-utf8](https://github.com/zarr-developers/zarr-python/blob/main/zarr/util.py#L47), although apparently Numpy arrays with `dtype="O"` are not supported at the moment (at least it throws an exception on my system when trying to write such an array to Zarr)

Reply:
Ok, then let's leave it like this and make this a breaking change. I think the storage size should generally be smaller if we use variable length strings.